Smartling.API/CloudLog/SmartlingCloudLogger.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Threading;
+using System.Threading.Tasks;
+using Smartling.Api.Model;
+using Smartling.Api.Threading;
+
+namespace Smartling.Api.CloudLog
+{
+  public class SmartlingCloudLogger : IDisposable
+  {
+    private const int BufferMaxSize = 100;
+    private const int TimeoutSeconds = 3;
+    private enum FlushMode { ForceFlush, DefaultFlush }
+    private Timer idleFlushTimer;
+    private const int DefaultLogThreads = 5;
+
+    private ConcurrentQueue<LoggingEventData> loggingEvents = new ConcurrentQueue<LoggingEventData>();
+
+    // We share the same thread pool across all instances
+    private static readonly LimitedConcurrencyLevelTaskScheduler scheduler = new LimitedConcurrencyLevelTaskScheduler(DefaultLogThreads);
+    private static readonly TaskFactory taskFactory = new TaskFactory(scheduler);
```

Comment:
Is it ok that it will be not threadsafe access to this object?

Reply:
According to doc https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.taskfactory?view=netframework-4.7.2#thread-safety `All public and protected members of TaskFactory are thread-safe and may be used concurrently from multiple threads.`